Add tool guide, getting started, python tutorial (#324)
terra-project/docs: docs/develop/get-started.md

## How to get started <img src="/img/icon_innovation.svg" height="40px">

Use this beginner-friendly guide to start building on Terra. 

:::{admonition} Learn about Terra
:class: note

> If you want to learn about how Terra works, Luna, validators, or governance, visit the [Terra protocol overview](../learn/protocol.md). 

Reviewer: ```suggestion
To learn more about about Terra concepts like Luna, validators, and governance, visit the [Terra protocol overview](../learn/protocol.md). 
```

:::

Follow the steps below to get started with Terrain and build your own dApp!

### 1. Set up Terrain

Terrain is a user-friendly development environment for developing smart contracts. Terrain makes it easy to create smart contract templates, deploy them to the Terra blockchain, and connect them to a frontend app.

Follow the [Terrain setup tutorial](./dapp/quick-start/initial-setup.md) to get started using Terrain. 

### 2. Download LocalTerra and start building

LocalTerra is a mini Terra blockchain that runs locally on your machine. You can use Terrain as a test environment for your projects. 

> When you start LocalTerra, you’ll see a single validator writing transactions to your mini blockchain. LocalTerra comes with [10 configurable wallet accounts](./how-to/localterra/accounts.md) to make testing easier. 

Reviewer: ```suggestion
LocalTerra is a scaled-down Terra blockchain that allows you to test your dApp locally. LocalTerra provides a single validator node and [10 configurable wallet accounts](./how-to/localterra/accounts.md). 
```

Follow the [Terrain with LocalTerra tutorial](./dapp/quick-start/using-terrain-localterra.md) to start using Terrain and create a simple counter app. 

### 3. Mint an NFT

> After getting familiar with Terrain and LocalTerra, you can get to the fun part: minting NFTs with Terrain. 

Reviewer: ```suggestion
Now that you are familiar with Terrain and LocalTerra, mint an NFT with Terrain. 
```

> The [NFT minting](./dapp/quick-start/mint-an-nft.md) tutorial walks you through the creation of an NFT minting application, complete with Terrain app scaffolding, smart contract usage, and a front end. You'll even get to mint your own NFT!

Reviewer: ```suggestion
The [NFT minting](./dapp/quick-start/mint-an-nft.md) tutorial walks you through the creation of an NFT minting application using the dApp scaffolding and front end provided by Terrain. 
```

I think it would be good to prune this sentence down. Not sure if my edit is the best way to go about it, but just an idea. Might also make sense  to consolidate with the top sentence to avoid redundancy.

:::{admonition} Terra Academy
:class: tip

> If you want to take interactive courses about programming in Rust, CosmWasm libraries, and developing your own smart contracts, visit [the Terra Academy](https://academy.terra.money)

Reviewer: ```suggestion
For interactive courses on developing smart contracts using Rust and CosmWasm libraries, visit [Terra Academy](https://academy.terra.money).
```

:::

## Development Suite

Congrats! You have the basics down! 

Now you can dive into the other tools in the Terra Development Suite:

### Terra<span/>.js

> If you are familiar with Javascript, use Terra<span/>.js to create bots, power NFT mints, and for all-purpose back end services. Follow the [Terra.js tutorial](./sdks/terra-js/getting-started.md) to get started.

Reviewer: ```suggestion
Use Terra<span/>.js to create bots, power NFT mints, and for all-purpose back end services. Follow the [Terra.js tutorial](./sdks/terra-js/getting-started.md) to get started.
```

Thought: I think we should remove the "If you are familiar with Javascript". The way I read this sentence, we are trying to emphasize that the JS library should be chosen if a dev wants to "create bots, power NFT mints, etc.". I think some folks might read the original sentence and assume the library should only be used by JS devs. Same logic with the Python library copy below

### Terra<span/>.py

>  If you are a Python developer, use Terra<span/>.py for on-chain analytics, Jupyter Notebooks, data processing, and other mathematical processes best suited for Python. Follow the [Terra.py](https://terra-money.github.io/terra.py/tutorial.html) tutorial to get started.

Reviewer: ```suggestion
Use Terra<span/>.py for on-chain analytics, Jupyter Notebooks, data processing, and other mathematical processes. Follow the [Terra.py](https://terra-money.github.io/terra.py/tutorial.html) tutorial to get started.
```

Ditto comment above

 
### Wallet Provider and Templates: 

If you have a back end for your app and want a front end to connect to Terra Station, use Wallet provider and Terra Templates. Follow the [Wallet Provider tutorial](./sdks/wallet-provider/wallet-provider-tutorial.md) to get started. 

### Other Tools

Terra also provides other tools for building:

- [Finder](https://finder.terra.money): Terra’s multi-purpose block explorer. 
- [Faucet](https://faucet.terra.money): Get tokens sent to your testnet address. 
- [`terrad`](./how-to/terrad/install-terrad.md): the node daemon and command line interface that connects to the Terra network. 

## Resources

### TerraWiki<span/>.org

> Check out the Terra Wiki for tutorials, reference material, and troubleshooting guides. Written by devs, for devs, [terrawiki.org](http://terrawiki.org) is the community documentation platform for all things Terra. Start browsing, or log in and create your own article.

Reviewer: ```suggestion
Written by devs, for devs, [terrawiki.org](http://terrawiki.org) is the community documentation platform for all things Terra, including community-generated tutorials, reference material, and troubleshooting guides. Start browsing, or log in and create your own article.
```

Just a suggestion. I like the overall vibe of this sentence

### Agora Developers' Forum

Join the [Developers' Forum](https://agora.terra.money/c/developer/20) and talk to other devs about code, projects, and troubleshooting.

### Terra Academy

> Take interactive courses and learn about coding in Rust, using CosmWasm libraries and developing, testing, and deploying your own smart contracts. Visit [the Terra Academy](https://academy.terra.money) site to get started.

Reviewer: A few thoughts here:
- Terra Academy is already mentioned in the callout above, so it might make sense to remove this section.
- However, if you want it there for redundancy, then I would suggest keeping the copy identical in each description for consistency

### Terra Discord Server

> If you would like to connect with the developer community and ask questions related to software development on Terra, join the [Terra Discord server](https://discord.com/invite/EuKCeGFb93).

Reviewer: ```suggestion
If you would like to connect with the developer community and ask questions related to software development on Terra, join the [Terra Discord server](https://discord.com/invite/EuKCeGFb93). Once you've joined the server:
```

This seems to be to be a procedure, so it might make sense to uses steps here i.e. once you join the server, do X Y and Z. See below comment as well

Pick up the `Developer Role` by clicking on the Green Bug icon `🪲` under the `🪲 | dev-roles` channel.

> Once you subscribe, a new section of developer channels should appear under `🪲 DEV` where you may select the relevant location to ask your questions and receive support.

Reviewer: ```suggestion
1.  Select `Developer Role` by clicking on the Green Bug icon `🪲` under the `🪲 | dev-roles` channel.

2. Once you are subscribed, a new section with the Terra developer channels appears under `🪲 DEV`. 

3. Select the relevant channel to ask your questions and receive support.
```

I think we should spell out the steps here. At a later time, might also make sense to list each channel and what that channel is generally used for, so that folks know where to go based on what questions they have, etc.
